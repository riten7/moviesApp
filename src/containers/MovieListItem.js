import React from 'react';
import { Col, Card } from 'antd';
import { Link } from 'react-router-dom';
import { PLACEHOLDER_IMG } from '../actions/Constant';

const MovieListItem = ({ movie }) => {
  const { Meta } = Card;

  return (
    <>
      <Col key={movie._id} className="gutter-row" span={4}>
        <Link to={{ pathname: `/movie/${movie.id}`}}>
        <Card hoverable
          style={{ width: 240 }}
          cover={<img src={movie.poster ? movie.poster : PLACEHOLDER_IMG} className="card-img-top" alt={movie.title} />}>
          <Meta title={movie.title} description={movie.type} />
        </Card>
        </Link>
      </Col>
    </>
  );
}

// function checkURL(url) {
//     return(url.match(/\.(jpeg|jpg|gif|png)$/) != null);
// }

export default MovieListItem;