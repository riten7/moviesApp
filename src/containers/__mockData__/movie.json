{
  "_id": "5e8070706eb3ec0e2f36712b",
  "title": "Test Case Url Update",
  "type": "Documentary",
  "rating": 3,
  "date": "2020/03/10",
  "poster": "https://m.media-amazon.com/images/M/MV5BMTU3NjE5NzYtYTYyNS00MDVmLWIwYjgtMmYwYWIxZDYyNzU2XkEyXkFqcGdeQXVyNzQzNzQxNzI@._V1_SY1000_SX677_AL_.jpg",
  "description": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.",
  "id": "burx2s81pa"
}