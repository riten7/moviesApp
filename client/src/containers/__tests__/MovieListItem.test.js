import React from "react";
import { mount } from "enzyme";
import MovieListItem from "../MovieListItem";
import { BrowserRouter } from 'react-router-dom';
import movie from '../__mockData__/movie.json';

describe("Component renders properly", () => {
  it("Test MovieListItem Value", () => {
    const wrapper = mount(<BrowserRouter><MovieListItem movie={movie} /></BrowserRouter>);
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.find("img").prop("src")).toEqual('https://m.media-amazon.com/images/M/MV5BMTU3NjE5NzYtYTYyNS00MDVmLWIwYjgtMmYwYWIxZDYyNzU2XkEyXkFqcGdeQXVyNzQzNzQxNzI@._V1_SY1000_SX677_AL_.jpg');
    expect(wrapper.find('[href="/movie/burx2s81pa"]').at(0).length).toBe(1);
    expect(wrapper.find('.ant-card-meta-description').text()).toEqual('Documentary');
  });
});