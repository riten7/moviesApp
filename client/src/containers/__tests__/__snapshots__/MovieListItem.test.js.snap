// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders properly Test MovieListItem Value 1`] = `"<div class=\\"ant-col ant-col-4 gutter-row\\"><a href=\\"/movie/burx2s81pa\\"><div style=\\"width: 240px;\\" class=\\"ant-card ant-card-bordered ant-card-hoverable\\"><div class=\\"ant-card-cover\\"><img src=\\"https://m.media-amazon.com/images/M/MV5BMTU3NjE5NzYtYTYyNS00MDVmLWIwYjgtMmYwYWIxZDYyNzU2XkEyXkFqcGdeQXVyNzQzNzQxNzI@._V1_SY1000_SX677_AL_.jpg\\" class=\\"card-img-top\\" alt=\\"Test Case Url Update\\"></div><div class=\\"ant-card-body\\"><div class=\\"ant-card-meta\\"><div class=\\"ant-card-meta-detail\\"><div class=\\"ant-card-meta-title\\">Test Case Url Update</div><div class=\\"ant-card-meta-description\\">Documentary</div></div></div></div></div></a></div>"`;
